### DISCUSSION TITLE
Exists validation not works properly with BelongsTo relationship and custom none primaryKey as ownerKey

### DISCUSSION THREAD

**Discussion by feeh27** [laravel/nova-issues#5019] [Q&A] [answered]
- Laravel Version: 9.38.0
- Nova Version: 4.17.1
- PHP Version: 8.1.1
- Database Driver & Version: MariaDB 10.5
- Operating System and Version: Windows 11
- Browser type and version: Google Chrome 106.0.5249.119
- Reproduction Repository: https://github.com/###/###

### Description:
When I try create a new register (Asset), the following message is returned: "The Entity field is required.".

Debugging I noticed that Laravel Nova sends the id for validation instead of the code field, which would be correct.

### Detailed steps to reproduce the issue on a fresh Nova installation:

- Create a new Asset using Laravel Nova interface
- Configure models and resources as below:

I have a two models, entity and assets:
![DER](https://user-images.githubusercontent.com/13547485/199807845-9505dc3a-d29b-43d5-a6a5-fec57e670555.jpeg)

In my assets Model I've entity method with BelongsTo relationship.

```php
public function entity(): BelongsTo
{
    return $this->belongsTo(
        Entity::class, // Eloquent model
        Asset::ENTITY_CODE, // Foreing key
        Entity::CODE, // Owner key
    );
}
```

In Entity Resource (Laravel Nova):
```php
public function fields(NovaRequest $request): array
{
    return [
        ID::make()->sortable(),
        Text::make('Code')->hideWhenUpdating()->sortable(),
        Text::make('Name')->sortable(),
        Text::make('Description')->sortable(),
        BelongsTo::make('Entity Code', null, Entity::class)
                ->dontReorderAssociatables()
                ->withoutTrashed()
                ->sortable()
                ->hideWhenUpdating()
                ->creationRules('exists:entities,code'),
    ];
}
```

**Comment by crynobone** [comment, marked as answer]
Hi there,

Unfortunately, this is something that we are not able to support at the moment for various limitations and since it may be solved with an upcoming feature I don't think it is worth it to complicate `BelongsTo` field at the moment.

In the meantime I believe you can use the following:

```php
BelongsTo::make('Entity Code', null, Entity::class)
  ->exceptOnForms()
  ->sortable(),

Select::make('Entity Code')
  ->options(function () {
    return \App\Models\Entity::query()->pluck('name', 'code');
  })->onlyOnForms()
  ->hideWhenUpdating()
  ->creationRules('exists:entities,code'),
```

**Reply by feeh27** [reply]
No problems, this workaround works for me.

Thanks.
